src/Repository/CommentRepository.php

```diff
@@ -49,11 +49,17 @@ public function getCommentsForActivity(Activity $activity): QueryBuilder
         return $queryBuilder;
     }
 
+    /**
+     * @param User $user
+     * @throws ORMException
+     * @throws OptimisticLockException
+     */
     public function anonymizeUserComments(User $user): void
     {
         $userComments = $this->findBy(array('user' => $user));
         foreach ($userComments as $userComment) {
             $userComment->setUser(null);
+            $this->save($userComment);
```

Comment:
Do an `$em->flush();` outside of foreach (instead of this save) so that all the updates are done via one query to the database.